pkg/codegen/internal/test/testdata/simple-enum-schema/dotnet/Enums.cs

```diff
@@ -0,0 +1,31 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System.Runtime.Serialization;
+
+namespace Pulumi.PlantProvider
+{
+    /// <summary>
+    /// plant container colors
+    /// </summary>
+    public enum ContainerColor
+    {
+        [EnumMember(Value = "red")]
+        Red,
+        [EnumMember(Value = "blue")]
+        Blue,
+        [EnumMember(Value = "yellow")]
+        Yellow,
```

Comment:
I'm going to dig into this more deeply, but my naive understanding is that using C# enums gets us the added goodness of Intellisense that we don't get with this kind of struct. Thanks for raising this though, I'll spend some time looking through the options here.

Reply:
> my naive understanding is that using C# enums gets us the added goodness of Intellisense that we don't get with this kind of struct

Would be curious to know the differences. From my initial check (VS Code, would be curious to know the difference on full VS on Windows).

Enum:

<img width="799" alt="Screen Shot 2020-10-13 at 3 32 00 PM" src="https://user-images.githubusercontent.com/710598/95923082-93f58380-0d69-11eb-95e7-090286beeb79.png">

Struct:

<img width="791" alt="Screen Shot 2020-10-13 at 3 33 11 PM" src="https://user-images.githubusercontent.com/710598/95923098-99eb6480-0d69-11eb-9263-a358037d9143.png">

Reply:
Oh neat! Yeah, will check it out in Visual Studio and also [Rider](https://www.jetbrains.com/rider/).

Reply:
@justinvp The background here is that we designed C# representation around Azure API. Two facts that make it a non-issue there:
- Adding enum values is [considered a breaking change in the API](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Breaking%20changes%20guidelines.md#allowed-values-for-an-enum-have-changed)
- Azure .NET SDK uses enums ([example](https://github.com/Azure/azure-sdk-for-net/blob/78cb37ff1ba9fb628ce8f053eaf6cec5320008a2/sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/StemmerTokenFilterLanguage.cs#L22))

However, your point makes sense in the global scope of different providers. I agree that we need to formalize this. Either add a number as hashcode indeed, or move to the struct-based approach (here is [another example](https://github.com/Azure/azure-sdk-for-net/blob/f775e1f8b8d33904015800c4d0894115a6d22515/sdk/storage/Azure.ResourceManager.Storage/src/Generated/Models/Kind.cs#L14) from the Azure SDK).

Reply:
@mikhailshilkov In fact, this is still an issue even for the azure provider/API. Adding enum values is only considered a breaking change in the API if they are "strict" enums, not so for "relaxed" enums. However, since we've decided to represent "relaxed" enums as just the union of the underlying and enum type, we are representing all enums using this same representation.

I'll move forward with the struct-based approach once I confirm IDE support. This would apply to enums of both string and number types, while int and bool can continue to be emitted as enums since they are naturally integers.

Reply:
Yeah, makes sense. If we use the structs, I wonder if we should simplify the relaxed enums from `Union<ContainerColor, string>` to a variant of `ContainerColor` with a public constructor and/or an implicit conversion from `string`.